shtab/__init__.py

```diff
@@ -421,8 +421,7 @@ def complete_bash(parser, root_prefix=None, preamble="", choice_functions=None):
     COMPREPLY=( $(compgen -W "${current_option_strings[*]}" -- "${completing_word}") )
   else
     # use choices & compgen
-    local IFS=$'\\n'
-    COMPREPLY=( $(compgen -W "${current_action_choices}" -- "${completing_word}") \\
+    COMPREPLY=( $(compgen -W "${current_action_choices[*]}" -- "${completing_word}") \\
```

Comment:
what if `current_action_choices = (foo 'bar baz')`? This would result in:

- `compgen -W "foo bar bat" -- b` => `bar` and `baz` separately (re-breaking #74)

solution: escape non-letters?

- `compgen -W "foo bar\ bat" -- b` => `bar baz`

Reply:
```suggestion
    COMPREPLY=( $(compgen -W "${current_action_choices[@]}" -- "${completing_word}") \\
```

well that was easy 😅

Reply:
ok finally fixes it I think:

```suggestion
    local IFS=$'\\n' # items may contain spaces, so delimit using newline
    COMPREPLY=( $([ -n "${current_action_compgen}" ] && "${current_action_compgen}" "${completing_word}") )
    unset IFS
    COMPREPLY+=( $(compgen -W "${current_action_choices[*]}" -- "${completing_word}") )
```